django/utils/numberformat.py

```diff
@@ -28,6 +28,11 @@ def format(number, decimal_sep, decimal_pos=None, grouping=0, thousand_sep='',
     sign = ''
     if isinstance(number, Decimal):
         str_number = '{:f}'.format(number)
+    elif isinstance(number, float):
+        # if decimal_pos is not given, render it with sufficient precision
+        precision = decimal_pos or 8
```

Comment:
What's magic about the value of 8? Seems like it can be changed to other nearby integer values without any tests failing.